source/common/common/version.cc

```diff
@@ -21,18 +25,59 @@ const std::string& VersionInfo::revisionStatus() {
 }
 
 const std::string& VersionInfo::version() {
+  CONSTRUCT_ON_FIRST_USE(std::string,
+                         fmt::format("{}/{}/{}/{}/{}", revision(), BUILD_VERSION_NUMBER,
+                                     revisionStatus(), buildType(), sslVersion()));
+}
+
+const envoy::api::v2::core::BuildVersion& VersionInfo::buildVersion() {
+  static envoy::api::v2::core::BuildVersion result = makeBuildVersion();
```

Comment:
This should probably be building a `envoy::api::v2::core::BuildVersion*` to avoid static destruction fiasco.

Reply:
done